storages/chat-store/migrations/2026-07-27-000000_message_arrival_order/up.sql

```
@@ -0,0 +1,18 @@
+-- time and is biased: this library stamps a constant `3EB0` prefix on every id
+-- it generates, while a peer's ids are effectively uniform hex, so descending
+-- id order put the inbound message on top for ~75% of ties — a reply rendering
+-- above the message it answers, every time.
+--
+-- The tiebreak is `rowid` now: the order the socket delivered the rows, which
+-- is the question the tiebreak was always asking. Realign the index to match.
+-- An ASC index scanned in reverse yields `timestamp_ms DESC, rowid DESC`
+-- directly (index keys carry the rowid as their trailing column), so the new
+-- sort streams from the index with no temp B-tree — the DESC form could not,
+-- because a forward scan of it gives `rowid ASC` within a tie.
+DROP INDEX IF EXISTS idx_messages_chat_time;
```

SQLFluff (4.2.2)

[error] DROP statement should use CONCURRENTLY to avoid locking the table.

(PG01)